homeassistant/components/light/wemo.py

```diff
@@ -140,3 +148,89 @@ def turn_off(self, **kwargs):
     def update(self):
         """Synchronize state with bridge."""
         self.update_lights(no_throttle=True)
+
+
+class WemoDimmer(Light):
+    """Representation of a WeMo dimmer."""
+
+    def __init__(self, device):
+        """Initialize the WeMo dimmer."""
+        self.wemo = device
+        self._brightness = None
+        self._state = None
+        # look up model name once as it incurs network traffic
+        self._model_name = self.wemo.model_name
+
+        wemo = get_component('wemo')
+        wemo.SUBSCRIPTION_REGISTRY.register(self.wemo)
+        wemo.SUBSCRIPTION_REGISTRY.on(self.wemo, None, self._update_callback)
+
+    def _update_callback(self, _device, _type, _params):
+        """Update the state by the Wemo device."""
+        _LOGGER.info("Subscription update for  %s", _device)
+        updated = self.wemo.subscription_update(_type, _params)
+        self._update(force_update=(not updated))
+
+        if not hasattr(self, 'hass'):
+            return
+        self.schedule_update_ha_state()
+
+    @property
+    def unique_id(self):
+        """Return the ID of this WeMo dimmer."""
+        return "{}.{}".format(self.__class__, self.wemo.serialnumber)
+
+    @property
+    def name(self):
+        """Return the name of the dimmer if any."""
+        return self.wemo.name
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return SUPPORT_BRIGHTNESS
+
+    @property
+    def brightness(self):
+        """Return the brightness of this light between 1 and 100."""
+        wemobrightness = int(self.wemo.get_brightness())
+        return int((wemobrightness * 255) / 100)
+
+    @property
+    def is_on(self):
+        """Return true if dimmer is on. Standby is on."""
+        return self._state
+
+    def update(self):
+        """Update WeMo state."""
+        self._update(force_update=True)
+
+    def _update(self, force_update=True):
+        """Update the device state."""
+        try:
+            self._state = self.wemo.get_state(force_update)
+            wemobrightness = int(self.wemo.get_brightness(force_update))
+            self._brightness = int((wemobrightness * 255) / 100)
+        except AttributeError as err:
+            _LOGGER.warning("Could not update status for %s (%s)",
+                            self.name, err)
+
+    def turn_on(self, **kwargs):
+        """Turn the dimmer on."""
+        self._state = WEMO_ON
+        self.wemo.on()
+
+        # Wemo dimmer switches use a range of [0, 100] to control
+        # brightness. Level 255 might mean to set it to previous value
+        if ATTR_BRIGHTNESS in kwargs:
+            self._brightness = kwargs[ATTR_BRIGHTNESS]
+            brightness = int((self._brightness / 255) * 100)
+        else:
+            brightness = 255
+        self.wemo.set_brightness(brightness)
```

Comment:
There's no direct update of home assistant state in the `turn_on` method. Is it not needed? Compare with the `turn_off` method.

Reply:
When the `self.wemo.set_brightness(brightness)` call is made, isn't the callback from wemo called in turn? If so, then you should not update state here, but late the callback do that. There's already a call to `self.schedule_update_ha_state()` in the callback. So if all this is true, then you should remove the `self.schedule_update_ha_state()` from `turn_on` and from `turn_off`.

Reply:
Removed these lines in my recent update as well